addons/mail/static/src/service_worker.js

```diff
@@ -1,5 +1,43 @@
 /* eslint-env serviceworker */
 /* eslint-disable no-restricted-globals */
+class FIFOMap {
+    constructor(maxSize) {
+        this.maxSize = maxSize;
+        this.map = new Map();
+        this.keys = [];
+    }
+
+    set(key, value) {
+        if (!this.map.has(key) && this.map.size >= this.maxSize) {
+            const oldestKey = this.keys.shift();
+            this.map.delete(oldestKey);
+        }
+        if (!this.map.has(key)) {
+            this.keys.push(key);
+        }
+        this.map.set(key, value);
+    }
+
+    get(key) {
+        return this.map.get(key);
+    }
+
+    has(key) {
+        return this.map.has(key);
+    }
+
+    delete(key) {
+        if (this.map.has(key)) {
+            this.map.delete(key);
+            const index = this.keys.indexOf(key);
+            if (index > -1) {
+                this.keys.splice(index, 1);
+            }
+        }
+    }
+}
+const channelFetched = new FIFOMap(20);
```

Comment:
I feel like the name is not explicit enough for a global variable.
Explicit name might a little bit long, but I'm thinking about `channelFetchedResponseByMessageId`